java/code/src/com/suse/manager/webui/controllers/DownloadController.java

```diff
@@ -174,8 +175,16 @@ public static Object downloadMetadata(Request request, Response response) {
             validateToken(token, channelLabel, filename);
         }
 
-        if (!file.exists() && filename.equals("comps.xml")) {
-            File compsFile = getCompsFile(ChannelFactory.lookupByLabel(channelLabel));
+        if (!file.exists()) {
+            // Check if a comps file is being requested and if we have it
+            File compsFile = null;
+            if (filename.equals("comps.xml")) {
+                compsFile = getCompsFile(ChannelFactory.lookupByLabel(channelLabel));
+            }
+            else if (filename.equals("modules.yaml")) {
+                compsFile = getModulesFile(ChannelFactory.lookupByLabel(channelLabel));
```

Comment:
The variable name compsFile is now missleading as it can also be modules.yaml.  Would an extra commit which renames it to mdFile (metadata file) be usefull?

Reply:
I was thinking about `rhnChannelComps` table when naming this. There, modules file is just another comps file with `type = 2`. But I guess `mdFile` is clearer in this case :+1: